device/src/io.rs

```diff
@@ -177,18 +173,28 @@ impl<'a, U> SerialIo<'a, U> {
         }
     }
 
-    pub fn write_bytes(&mut self, words: &[u8]) -> Result<(), SerialInterfaceError>
+    pub fn write_bytes(&mut self, bytes: &[u8]) -> Result<(), SerialInterfaceError>
     where
         U: uart::Instance,
     {
         match self {
             SerialIo::Jtag(jtag) => jtag
-                .write_bytes(words)
-                .map_err(|_| SerialInterfaceError::JtagError),
-            SerialIo::Uart(uart) => uart
-                .write_bytes(words)
-                .map_err(SerialInterfaceError::UartWriteError),
+                .write_bytes(bytes)
+                .map_err(|_| SerialInterfaceError::JtagError)?,
+            SerialIo::Uart(uart) => {
+                for byte in bytes {
+                    while let Err(e) = uart.write(*byte) {
```

Comment:
> you can end up writing less than the bytes and won't notice it

Wouldn't this have previously returned an error coming from `uart.write_bytes`?

The improvement I see here is not erroring on a `WouldBlock` and instead trying again.

Reply:
My guess is this is what the previous implementation of `write_bytes` did: Keep writing and trying again if you get `WouldBlock` -- now it just stops on `WouldBlock` and returns the number of bytes written.